docs/root/configuration/listeners/network_filters/connection_limit_filter.rst

```diff
@@ -0,0 +1,96 @@
+.. _config_network_filters_connection_limit:
+
+Connection Limit Filter
+=======================
+
+Background
+----------
+
+Network connections are a limited resource that we need some functionality to protect. Envoy has the capability to limit the rate of new connections via the L4 `local rate limit filter <https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/network_filters/local_rate_limit_filter>`_. It would be useful to be able to limit the number of connections on a filter chain basis or based on some descriptors from the request.
+
+Goals
+-----
+
+1. Protection for resources such as connections, CPU, memory, etc. by making sure every filter chain gets fair share of connection resources.
+2. Preventing any single entity based on filter chain match or descriptors from consuming a large number of connections to ensure fair share of the connections.
+
+Filter Overview
+---------------
+
+We are planning to add a new Connection Limit Filter in Envoy since there’s no existing connection limit functionality per filter chain in Envoy. It will be a L4 filter that is designed to be configurable and extensible. This way we can prevent any one entity from taking up too much network connections resource.
```

Comment:
The use of future tenses is a bit excessive for a user doc. Could you please edit the text to be a manual for the filter?

Reply:
Updated the doc, please take a look.